src/main/java/com/redhat/devtools/lsp4ij/features/formatting/LSPClientSideOnTypeFormattingTypedHandler.java

```diff
@@ -50,120 +52,112 @@ public Result charTyped(char charTyped,
                             @NotNull Project project,
                             @NotNull Editor editor,
                             @NotNull PsiFile file) {
-        LSPFormattingFeature formattingFeature = getFormattingFeature(file);
-        if (formattingFeature != null) {
-            boolean rangeFormattingSupported = formattingFeature.isRangeFormattingSupported(file);
+        // Gather all of the relevant client configuration
+        Ref<Boolean> rangeFormattingSupportedRef = Ref.create(false);
+        ClientSideOnTypeFormattingSettings onTypeFormattingSettings = new ClientSideOnTypeFormattingSettings();
+        LanguageServiceAccessor.getInstance(project).processLanguageServers(
+                file,
+                ls -> {
+                    // Only include servers that support formatting and don't support server-side on-type formatting
+                    LSPClientFeatures clientFeatures = ls.getClientFeatures();
+                    LSPFormattingFeature formattingFeature = clientFeatures.getFormattingFeature();
+                    LSPOnTypeFormattingFeature onTypeFormattingFeature = clientFeatures.getOnTypeFormattingFeature();
+                    if (formattingFeature.isEnabled(file) && formattingFeature.isSupported(file) &&
```

Comment:
All of these feature flags default to `false`/disabled, so if we don't find a language server that supports formatting but not server-side on-type formatting, everything below will be an effective no-op.